### PR TITLE
CTCP msg impl

### PR DESCRIPTION
**Pull request by kropptrevor**

### CODE REVIEW

**Comment by kroppt** [comment]
I'm starting to think there's something wrong with the way we designed this.
`encoded, err := Plain{"text"}.Encode()` provides no particular information related to CTCP, yet that's what the text gets encoded to.

I think when the package was changed from `ctcp` to `msg` something was lost about the functionality of these methods and functions. Whether we have to change the purpose of the structs or change names, I'm not sure. I will have to think about it when I have time this weekend.

**Comment by kroppt** [comment]
We should not be casting runes as bytes for encoding/decoding.
https://play.golang.org/p/iq_E7lK4JSB
A rune in a string can take up several bytes, each of which needs normal escaping.

File: msg/msg.go [hunk 1]
```diff
@@ -39,6 +75,110 @@ func (c CTCP) Value() string {
...
+	case xQuote:
+		return string(xQuote)
+	default:
+		return ""
+	}
+}
+
+func isWhitespace(char byte) bool {
+	switch char {
+	case '\001':
```
Comment (kroppt) [inline msg/msg.go]: should be `xDelim`

File: msg/msg.go [hunk 2]
```diff
@@ -39,6 +75,110 @@ func (c CTCP) Value() string {
 }
 
 // Decode turns the input message from a CTCP escaped message into plain text.
-func Decode(string) ([]Valuer, error) {
-	return nil, errors.New("not implemented")
+func Decode(message string) ([]Valuer, error) {
```
Comment (kroppt) [inline msg/msg.go:82]: If it accepts a CTCP-escaped message, why not use CTCP as the message's type, in which case it probably should be a method? Or change this comment to be more accurate.

**Comment by kroppt** [comment]
Based on what this does, it seems like we don't need a separate parser, is that correct? If so, we need to start adding other things like tags, prefixes, etc.
Otherwise, please explain the use case there.

File: msg/msg.go (hunk 2 above)
Comment (kropptrevor) [reply msg/msg.go:82]: It has been updated

**Pull request closed without merging**